sycl/source/detail/jit_compiler.cpp

```diff
@@ -1243,7 +1259,9 @@ sycl_device_binaries jit_compiler::compileSYCL(
 
   std::string FinalSource = ss.str();
 
-  std::string SYCLFileName = CompilationID + ".cpp";
+  // The filename must be stable, because it is part of the preprocessed output
+  // and in consequence, the cache key.
+  std::string SYCLFileName = "rtc.cpp";
```

Comment:
this is unfortunate. Some day in the future when debugging support is required, this "rt.cpp" filename for everything might become a problem. That day is likely no day soon, so I'm fine with it.   
Is there no other (simple) way for a file name that is unique for the kernel, but consistent between invocations (cache read and write )?

Reply:
Good point, maybe a short hash of the source string only? Or we could come back to the idea of letting the user set the filename (or a generic tag) via a property, which might be more useful for debugging.

In any case, let me check again how much effort it would be to keep the filename out of the cache key calculation.

Reply:
Disabling the line markers in the preprocessor output does the trick for now, so I reverted back to the original behaviour, using `rtc_<n>.cpp` as file name.